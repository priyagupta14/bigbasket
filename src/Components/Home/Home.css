.product-card {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}
@media screen and (max-width: 24rem) {
  .product-card{
    margin: auto;
    display: flex;
    justify-content: center;
  }
}
