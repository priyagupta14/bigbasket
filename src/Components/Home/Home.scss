@import '../../styles/index.scss';
.product-card {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}
@media screen and (max-width: $breakpoint-for-phone) {
  .product-card{
    margin: auto;
    display: flex;
    justify-content: center;
  }
  h1{
    font-size: large;
  }
}
