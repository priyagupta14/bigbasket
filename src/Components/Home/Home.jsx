/* eslint-disable react/prop-types */
/* eslint-disable react/require-default-props */
import React from 'react';
import PropTypes from 'prop-types';
import Product from '../Product/Product';
import './Home.scss';

const Home = (props) => {
  const allProducts = props.productList.map((eachProduct) => (
    <Product
      key={eachProduct.id}
      product={eachProduct}
      onIncrement={props.onIncrement}
      onDecrement={props.onDecrement}
    />
  ));
  return (
    <div>
      <h1> Fruits&Vegetables </h1>
      <div className="product-card">{allProducts}</div>
    </div>
  );
};

Home.propTypes = {
  productList: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      price: PropTypes.number.isRequired,
      count: PropTypes.number.isRequired,
      url: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }),
  ),
};
export default Home;
