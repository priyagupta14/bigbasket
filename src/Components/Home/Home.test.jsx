/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import {
  fireEvent, render, screen,
} from '@testing-library/react';
import Home from './Home';

describe(Home.name, () => {
  const mockProps = {
    categorizedProduct:
    {
      abc: [{
        category: 'abc',
        count: 2,
        id: 5,
        inCartCount: 0,
        name: 'milk',
        price: 10,
      },
      ],
      bcd:
        [
          {
            category: 'bcd',
            count: 2,
            id: 5,
            inCartCount: 0,
            name: 'apple',
            price: 10,
          },
        ],

    },
    onIncrement: jest.fn(),
    onDecrement: jest.fn(),
  };
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should match the snapshot', () => {
    const { container } = render(<Home {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
  it('should display header', () => {
    render(<Home {...mockProps} />);
  });
});
