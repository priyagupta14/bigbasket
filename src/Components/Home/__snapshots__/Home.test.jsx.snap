// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
<div>
  <div>
    <div>
      <h1>
        abc
      </h1>
      <div
        class="product-card"
      >
        <div
          class="product-card-element"
        >
          <img
            alt="fruitItem"
            data-testid="product-img"
          />
          <p>
            milk
          </p>
          <div
            class="price-counter"
          >
            <p>
              MRP
              10
              /-
            </p>
            <div
              class="zeroProduct"
            >
              <button
                type="button"
              >
                        +      
              </button>
              <p>
                0
                 
                in basket
              </p>
              <button
                type="button"
              >
                -
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <h1>
        bcd
      </h1>
      <div
        class="product-card"
      >
        <div
          class="product-card-element"
        >
          <img
            alt="fruitItem"
            data-testid="product-img"
          />
          <p>
            apple
          </p>
          <div
            class="price-counter"
          >
            <p>
              MRP
              10
              /-
            </p>
            <div
              class="zeroProduct"
            >
              <button
                type="button"
              >
                        +      
              </button>
              <p>
                0
                 
                in basket
              </p>
              <button
                type="button"
              >
                -
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
